lib/pool.js

```diff
@@ -159,11 +166,18 @@ class Pool extends EventEmitter {
 
       if (!client) {
         if (!this[kConnections] || this[kClients].length < this[kConnections]) {
-          client = new Client(this[kUrl], this[kOptions])
+          const options = { ...this[kOptions] }
+          options.tls = addCachedSessionToTLSOptions(this, options)
+
+          client = new Client(this[kUrl], options)
             .on('drain', this[kOnDrain])
             .on('connect', this[kOnConnect])
             .on('disconnect', this[kOnDisconnect])
 
+          if (!options.tls || (options.tls.reuseSessions !== false && !options.tls.session)) {
```

Comment:
```suggestion
          if (!options.tls || (options.tls.reuseSessions && !options.tls.session)) {
```

Reply:
The same comment as above regarding the default behavior applies here too.